dialect.go

```diff
@@ -13,4 +13,11 @@ type Dialect interface {
 	EncodeBytes(b []byte) string
 
 	Placeholder(n int) string
+
+	UpdateStmts() (string, string)
```

Comment:
could you comment on why each is added?

Reply:
- `UpdateStmts` was added to know what is the staement that has to be used to run UPDATEs. Unfortunatelly clickhosue is different in this part, it's in the shape of `ALTER TABLE ... UPDATE key = value` compared to others where it's `UPDATE ... SET key = value`
- `OnConflict`  adds support to add actions for constraint violation, e.g UPSERT. This works in postgres, mysql, etc. I can maybe add this change in a separate PR if you think it would be better to review, let me know if you want me to split this into smaller PRs, happy to do.
- `Proposed` is used with OnConflict as reference to proposed value in on conflict clause. You can see the TestOnConflict to see how it works.
- `SupportsOn` was added for clickhouse explicitly, but I removed it for now, I can maybe add it later if we really need to use it, and I think it should be done in a different way.

Reply:
Created a separate PR for the CI updates https://github.com/gocraft/dbr/pull/265 I can then split this into more pieces if that's ok.